src/Compilers/CSharp/Portable/Symbols/Metadata/PE/DynamicTypeDecoder.cs

```diff
@@ -349,25 +349,12 @@ private PointerTypeSymbol TransformPointerType(PointerTypeSymbol pointerType)
 
             var sig = type.Signature;
 
-            var transformedReturn = handle(ref this, sig.RefKind, sig.RefCustomModifiers, sig.ReturnTypeWithAnnotations);
-            if (transformedReturn is null)
+            var (transformedReturnWithAnnotations, madeChanges) = handle(ref this, sig.RefKind, sig.RefCustomModifiers, sig.ReturnTypeWithAnnotations);
+            if (transformedReturnWithAnnotations is null)
```

Comment:
`handle()` could return `TypeWithAnnotations` rather than `TypeWithAnnotations?` and the caller could use `if (transformedReturnWithAnnotations.IsDefault)`.